snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -737,6 +738,28 @@ def _apply_passthrough(
         section.update(passthrough)
         return bool(passthrough)
 
+    def validate_common_ids(self) -> None:
+        if (
+            not self._extracted_metadata
+            or not self._extracted_metadata.common_id_list
+            or "apps" not in self._config_data
+        ):
+            return
+
+        common_id_list = self._extracted_metadata.common_id_list
+        for app in self._config_data["apps"]:
+            app_common_id = self._config_data["apps"][app].get("common-id")
+            if app_common_id not in common_id_list:
+                app_name = _get_app_name_from_common_id(
+                    self._config_data, app_common_id
+                )
+                logger.warning(
+                    "Common ID {common_id!r} specified in app {app!r} is "
+                    "not used in any appstream metafile.".format(
```

Comment:
this is very specific to appstream, can you generalize the message?

Reply:
Message changed.